From 91a823edc714105c8900632147cebfc176ca003d Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Sat, 3 Sep 2022 19:06:22 +0200
Subject: [PATCH] macOS 10.7: Switch to OpenSSL

macOS before 10.8 requires to use OpenSSL because a lot of things is
simple missed.
---
 vendor/crypto-hash/Cargo.toml | 4 ++--
 vendor/crypto-hash/src/lib.rs | 8 ++++----
 vendor/libgit2-sys/build.rs   | 2 --
 3 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/vendor/crypto-hash/Cargo.toml b/vendor/crypto-hash/Cargo.toml
index 2ae218e617..37cd476d80 100644
--- a/src/vendor/crypto-hash/Cargo.toml
+++ b/src/vendor/crypto-hash/Cargo.toml
@@ -23,10 +23,8 @@
 repository = "https://github.com/malept/crypto-hash"
 [dependencies.hex]
 version = "0.3"
-[target."cfg(not(any(target_os = \"windows\", target_os = \"macos\")))".dependencies.openssl]
+[target."cfg(not(target_os = \"windows\"))".dependencies.openssl]
 version = "0.10"
-[target."cfg(target_os = \"macos\")".dependencies.commoncrypto]
-version = "0.2"
 [target."cfg(target_os = \"windows\")".dependencies.winapi]
 version = "0.3"
 features = ["minwindef", "wincrypt"]
diff --git a/vendor/crypto-hash/src/lib.rs b/vendor/crypto-hash/src/lib.rs
index fd553ba528..957756c21c 100644
--- a/src/vendor/crypto-hash/src/lib.rs
+++ b/src/vendor/crypto-hash/src/lib.rs
@@ -31,8 +31,7 @@
 //! By operating system:
 //!
 //! * Windows: `CryptoAPI`
-//! * Mac OS X: `CommonCrypto`
-//! * Linux/BSD/etc.: `OpenSSL`
+//! * Linux/BSD/Mac OS X etc.: `OpenSSL`
 //!
 //! # Supported Algorithms
 //!
@@ -43,10 +42,8 @@
 
 #![warn(missing_docs)]
 
-#[cfg(target_os = "macos")]
-extern crate commoncrypto;
 extern crate hex;
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
+#[cfg(not(target_os = "windows"))]
 extern crate openssl;
 #[cfg(target_os = "windows")]
 extern crate winapi;
@@ -59,7 +56,7 @@
 #[cfg(target_os = "windows")]
 #[path = "imp/cryptoapi.rs"]
 mod imp;
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
+#[cfg(not(target_os = "windows"))]
 #[path = "imp/openssl.rs"]
 mod imp;
 
